content/blog/origins-of-graphql.md

```diff
@@ -0,0 +1,62 @@
++++
+title = "Origins of Graphql"
+slug = "origin-of-graphql"
+date = 2023-02-23T11:33:15+05:30
+image = "/images/2023/02/graphql.png"
+draft = false
+authors = ["Nitin Rajkumar"]
+description = "Origins of graphql and what took so long for this approach to take its implementation"
```

Comment:
Correct me if I am wrong but I don't see any mention/answer of "what took so long" section!

Reply:
I mean I see the section for that but that section has just the history of communication protocals, not "what too so long"

For example, COVID vaccine is mRNA vaccine, which is a newer type of vaccine; Covid is probably the first example of that type of vaccine being using in a wide scale;
What took so long for it to be used have answers as "lack of urgency, no motivation to change already existing vaccination system, lack of technology resource, etc"

Reply:
Thanks for bringing this up. This section is mostly an opinion - I believe not until the advent of mobile applications  a new client unlike browser there was an inspiration or a need to prevent under fetching and over fetching of data from the server. I will try rewrite and present that section with evolutionary challenges of the protocols conveying "What were shortcomings of the each protocol and how it's advancement provides a solution for it". RPC->SOAP->REST->GQL. Would like to know your opinion on it too.